backend/api/Areas/Research/Controllers/ResearchFileController.cs

```diff
@@ -58,6 +59,106 @@ public IActionResult GetResearchFile(long id)
+                Description = "GENERAL Activity",
+                ActivityTemplateId = 1,
+                ActivityTemplateTypeCode = new Models.TypeModel<string>()
+                {
+                    Id = "GENERAL",
+                    Description = "General Visit ",
+                },
+                ActivityStatusTypeCode = new Models.TypeModel<string>()
+                {
+                    Id = "Draft",
+                    Description = "Draft",
+                },
+            });
+            activities.Add(new ActivityInstanceModel()
+            {
+                Id = 2,
+                Description = "Site Visit Activity",
+                ActivityTemplateId = 2,
+                ActivityTemplateTypeCode = new Models.TypeModel<string>()
+                {
+                    Id = "SITEVIS",
+                    Description = "Site Visit ",
+                },
+                ActivityStatusTypeCode = new Models.TypeModel<string>()
+                {
+                    Id = "Draft",
+                    Description = "Draft",
+                },
+            });
+            activities.Add(new ActivityInstanceModel()
+            {
+                Id = 3,
+                Description = "Survey Activity",
+                ActivityTemplateId = 3,
+                ActivityTemplateTypeCode = new Models.TypeModel<string>()
+                {
+                    Id = "SURVEY",
+                    Description = "Site Visit ",
+
+                },
+                ActivityStatusTypeCode = new Models.TypeModel<string>()
+                {
+                    Id = "Draft",
+                    Description = "Draft",
+                },
+            });
+            return new JsonResult(activities);
+        }
+
+
+        /// <summary>
+        /// Get the document types.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("activity-templates")]
+        [HasPermission(Permissions.ResearchFileView)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(List<ActivityTemplateModel>), 200)]
+        [SwaggerOperation(Tags = new[] { "activity-templates" })]
+        public IActionResult GetActivityTemplateTypes()
+        {
+            var activityTemplates = _pimsService.ActivityService.GetAllActivityTemplates();
+            var mappedActivityTemplates = _mapper.Map<List<ActivityTemplateModel>>(activityTemplates);
+            return new JsonResult(mappedActivityTemplates);
+        }
+
+        /// <summary>
+        /// Add the specified research file activity.
+        /// </summary>
+        /// <returns></returns>
+        [HttpPost("activity")]
+        [HasPermission(Permissions.ResearchFileAdd)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(ActivityInstanceModel), 200)]
+        [SwaggerOperation(Tags = new[] { "activityInstance" })]
+        public IActionResult AddActivityInstance(ActivityInstanceModel activityInstanceModel)
```

Comment:
I would recommend moving this one to its own controller as well given the resource type is an activity instance and not a research file

Reply:
As per developer notes in the story https://jira.th.gov.bc.ca/browse/PSP-3443 , this needs to be created in ResearchFile Controller